daemon/graphdriver/overlay2/overlay.go

```diff
@@ -515,7 +515,9 @@ func (d *Driver) Remove(id string) error {
 	dir := d.dir(id)
 	lid, err := ioutil.ReadFile(path.Join(dir, "link"))
 	if err == nil {
-		if err := os.RemoveAll(path.Join(d.home, linkDir, string(lid))); err != nil {
+		if len(lid) == 0 {
+			logrus.Errorf("refusing to remove empty link for layer %v", id)
+		}else if err := os.RemoveAll(path.Join(d.home, linkDir, string(lid))); err != nil {
```

Comment:
Looks like there's a gofmt issue here; 

```
03:37:39 Congratulations!  All commits are properly signed with the DCO!
03:40:00 daemon/graphdriver/overlay2/overlay.go:1::warning: file is not gofmted with -s (gofmt)
03:40:00 daemon/graphdriver/overlay2/overlay.go:1::warning: file is not goimported (goimports)
```

There should be a space after `}`;

```go
} else if err := os.RemoveAll(path.Join(d.home, linkDir, string(lid))); err != nil {
```